src/main/java/gregtech/api/capability/impl/MultiblockRecipeLogic.java

```diff
@@ -251,43 +249,37 @@ protected boolean prepareRecipeDistinct(Recipe recipe) {
     }
 
     @Override
-    protected int[] runOverclockingLogic(@Nonnull IRecipePropertyStorage propertyStorage, int recipeEUt, long maxVoltage, int recipeDuration, int amountOC) {
+    protected void modifyOverclockPre(@Nonnull int[] values, @Nonnull IRecipePropertyStorage storage) {
+        super.modifyOverclockPre(values, storage);
+
         // apply maintenance penalties
         Tuple<Integer, Double> maintenanceValues = getMaintenanceValues();
 
-        int[] overclock = null;
-        if (maintenanceValues.getSecond() != 1.0)
-
-            overclock = standardOverclockingLogic(
-                    Math.abs(recipeEUt),
-                    maxVoltage,
-                    (int) Math.round(recipeDuration * maintenanceValues.getSecond()),
-                    amountOC,
-                    getOverclockingDurationDivisor(),
-                    getOverclockingVoltageMultiplier()
-            );
+        // duration bonus
+        if (maintenanceValues.getSecond() != 1.0) {
+            values[1] = (int) Math.round(values[1] * maintenanceValues.getSecond());
+        }
+    }
 
-        if (overclock == null)
-            overclock = standardOverclockingLogic(
-                    Math.abs(recipeEUt),
-                    maxVoltage,
-                    recipeDuration,
-                    amountOC,
-                    getOverclockingDurationDivisor(),
-                    getOverclockingVoltageMultiplier()
-            );
+    @Override
+    protected void modifyOverclockPost(int[] overclockResults, @Nonnull IRecipePropertyStorage storage) {
+        super.modifyOverclockPost(overclockResults, storage);
 
-        if (maintenanceValues.getFirst() > 0)
-            overclock[1] = (int) (overclock[1] * (1 + 0.1 * maintenanceValues.getFirst()));
+        // apply maintenance penalties
+        Tuple<Integer, Double> maintenanceValues = getMaintenanceValues();
 
-        return overclock;
+        // duration penalty
+        if (maintenanceValues.getFirst() > 0) {
```

Comment:
According to your PR statement, the EU changes were supposed to be made post overclock, not the duration changes.  Is that still accurate and this logic is wrong? As currently you are modifiying the duration pre and post overclock, and not touching the EU

Reply:
EU was a misinterpretation on my part. I updated the PR description to reflect this. The changes made pre-oc are the bonus duration from the configurable maintenance hatch. The changes post-oc are the duration penalties from unfixed maintenance problems.